airline-web/app/controllers/ChatApplication.scala

```diff
@@ -17,9 +21,19 @@ class ChatApplication @Inject()(actorSystem: ActorSystem) extends Controller {
    Specifies how to wrap an out-actor that will represent
    WebSocket connection for a given request.
   */
-  def chatSocket = WebSocket.acceptWithActor[String, String] {
-    (request: RequestHeader) =>
-    (out: ActorRef) =>
-    Props(new ClientActor(out, chat))
-  }	 
+ // def chatSocket = WebSocket.acceptWithActor[String, String] {
+ //   (request: RequestHeader) =>
+ //   (out: ActorRef) =>
+ //   Props(new ClientActor(out, chat))
+ //  }	 
+   def chatSocket = WebSocket.tryAcceptWithActor[String, String] { request => 
+    Future.successful(request.session.get("userId") match {
```

Comment:
Wondering where does the userId come from? Is it authenticated :) ?

Reply:
I copied that from your code .. you tell me! But it's the authenticated ID

Reply:
ah ok it's from `AuthenticationObject` 😅